packages/mocktail/lib/src/mocktail.dart

```diff
@@ -1,664 +1,729 @@
-import 'dart:math';
+import 'dart:async';
 
+import 'package:collection/collection.dart';
 import 'package:matcher/matcher.dart';
-
-/// {@template mocktail_failure}
-/// An exception thrown from the mocktail library.
-/// {@endtemplate}
-class MocktailFailure implements Exception {
-  /// {@macro mocktail_failure}
-  const MocktailFailure(this.message);
-
-  /// The failure message
-  final String message;
-
-  @override
-  String toString() => 'MocktailFailure: $message';
+// ignore: deprecated_member_use
+import 'package:test_api/test_api.dart';
+
+import 'fake.dart';
+
+part '_arg_matcher.dart';
+part '_invocation_matcher.dart';
+part '_is_invocation.dart';
+part '_real_call.dart';
+part '_register_matcher.dart';
+part '_time_stamp_provider.dart';
+
+_WhenCall? _whenCall;
+_UntilCall? _untilCall;
+var _whenInProgress = false;
+bool _untilCalledInProgress = false;
+var _verificationInProgress = false;
+var _numMatchers = 0;
+
+final _timer = _TimeStampProvider();
+final _capturedArgs = <dynamic>[];
+final _storedArgs = <ArgMatcher>[];
+final _storedNamedArgs = <String, ArgMatcher>{};
+final _verifyCalls = <_VerifyCall>[];
+
+/// Opt-into [Mock] throwing [NoSuchMethodError] for unimplemented methods.
+///
+/// The default behavior when not using this is to always return `null`.
+void throwOnMissingStub(
```

Comment:
I've always found it weird that mockito in dart returns null for unstubbed method calls on Mocks.

I forget if that's the Java behavior, maybe that's where that comes from, I suspect it is because I have vague memories of debugging NREs in Java in my tests because of missing stubs.

Other languages I've used don't take that approach: they require you to stub all methods you intend to call otherwise they throw. 

I'm wondering is if you have any interest in adding an option to make all Mocks throw on unstubbed calls? And I'm also wondering if that might remove the need for the fallback values for user types?

Reply:
Oh I think I figured out that the answer to my second question is no. You'd still need the fallback value for parameters, right?

Reply:
Yup and you can use throwOnMissingStub https://github.com/felangel/mocktail/blob/f5ac82ed45e0791eb6552414f674fab79bfd378a/packages/mocktail/lib/src/mocktail.dart#L33 👍

Reply:
Oh, yea I see I can use that method on an individual mock instance, but I'm just wondering if you'd be open to a global setting to default to that behavior instead of having to call it each time I set up a mock.

Reply:
Yeah I'm definitely in favor of that but it would be a breaking change for those migrating from mockito which is why I maintained the same behavior. wdyt?

Reply:
Ah yea. I wouldn't want to force everyone to have to change strategy just to migrate. But I could imagine making it a globally set table option that defaults to false. Like `Mocktail.throwOnUndefinedStub = true`. 

It could default to false OR if you want this to be the intended way to use the library then you could default it to true and encourage folks migrating from Mockito to start by turning it off.

Reply:
You can do `Mock.throwOnMissingStub()` to configure this globally as part of v0.1.0 🎉